src/test/java/org/mule/tools/maven/dependency/model/ClassLoaderModelTest.java

```diff
@@ -0,0 +1,160 @@
+    new ClassLoaderModel(null, null);
+  }
+
+  @Test
+  public void validatePluginsTest() throws URISyntaxException {
+    expectedException.expect(IllegalArgumentException.class);
+    String notMulePluginsInitialMessage = "The following dependencies are not mule plugins but are trying to be added as such: ";
+    expectedException.expectMessage(notMulePluginsInitialMessage);
+
+    Set<Dependency> mulePlugins = buildDependencySet(NUMBER_MULE_PLUGINS, MULE_PLUGIN_CLASSIFIER);
+    Set<Dependency> notMulePlugins1 =
+        buildDependencySet(NUMBER_NOT_MULE_PLUGINS_DEPENDENCIES, NOT_MULE_PLUGIN_EXAMPLE_CLASSIFIER);
+    Set<Dependency> notMulePlugins2 = buildDependencySet(NUMBER_DEPENDENCIES_WITHOUT_CLASSIFIER, null);
+
+    Set<Dependency> allDependenciesCandidatesAsMulePlugins = new HashSet<>();
+
+    allDependenciesCandidatesAsMulePlugins.addAll(mulePlugins);
+    allDependenciesCandidatesAsMulePlugins.addAll(notMulePlugins1);
+    allDependenciesCandidatesAsMulePlugins.addAll(notMulePlugins2);
+
+    ClassLoaderModel model = new ClassLoaderModel(VERSION, buildArtifactCoordinates(MULE_APPLICATION_CLASSIFIER));
+
+    // 2 (NUMBER_MULE_PLUGINS) dependencies are mule plugins; the remaining 6 (NUMBER_NOT_MULE_PLUGINS_DEPENDENCIES +
+    // NUMBER_DEPENDENCIES_WITHOUT_CLASSIFIER) are not
+    model.validatePlugins(allDependenciesCandidatesAsMulePlugins);
+  }
+
+  @Test
+  public void addMulePluginTest() throws URISyntaxException {
+    Dependency mulePlugin1 = buildDependency(MULE_PLUGIN_CLASSIFIER);
+    Dependency mulePlugin2 = buildDependency(MULE_PLUGIN_CLASSIFIER);
+
+    SortedSet<Dependency> mulePlugin1Dependencies = new TreeSet<>(buildDependencySet(CONSTANT_NUMBER_DEPENDENCIES, null));
+    SortedSet<Dependency> mulePlugin2Dependencies = new TreeSet<>(buildDependencySet(INITIAL_NUMBER_DEPENDENCIES, null));
+
+    SortedMap<Dependency, SortedSet<Dependency>> mulePlugins = new TreeMap<>();
+    mulePlugins.put(mulePlugin1, mulePlugin1Dependencies);
+    mulePlugins.put(mulePlugin2, mulePlugin2Dependencies);
+
+    ClassLoaderModel model = new ClassLoaderModel(VERSION, buildArtifactCoordinates(MULE_APPLICATION_CLASSIFIER));
+
+    model.setMulePlugins(mulePlugins);
+
+    SortedSet<Dependency> additionalMulePlugin2Dependencies =
+        new TreeSet<>(buildDependencySet(ADDITIONAL_NUMBER_DEPENDENCIES, null));
+
+    model.addMulePlugin(mulePlugin2, additionalMulePlugin2Dependencies);
+
+    SortedMap<Dependency, SortedSet<Dependency>> actualMulePlugins = model.getMulePlugins();
+
+    assertThat("Number of mule plugins in set is not the expected", actualMulePlugins.keySet().size(),
+               equalTo(NUMBER_MULE_PLUGINS));
+    assertThat("Mule plugins set is not the expected", actualMulePlugins.keySet(), containsInAnyOrder(mulePlugin1, mulePlugin2));
+
+    assertThat("Number of dependencies of mulePlugin1 is not the expected", actualMulePlugins.get(mulePlugin1).size(),
+               equalTo(CONSTANT_NUMBER_DEPENDENCIES));
```

Comment:
I wouldn't put this CONSTANT_NUMBER_DEPENDENCIES as a constant as it seems to change from test to test it could be misleading